x-pack/plugins/ml/public/application/explorer/add_to_dashboard_control.tsx

```diff
@@ -0,0 +1,339 @@
+                    aria-label="Quick add"
+                    disabled={addedDashboards[item.id] === 'success'}
+                  />
+                </EuiToolTip>
+              );
+            },
+          },
+          {
+            name: i18n.translate('xpack.ml.explorer.dashboardsTable.editInDashboard', {
+              defaultMessage: 'Edit in dashboard',
+            }),
+            onClick: async (item) => {
+              // TODO generate an edit URL with predefined state instead (https://github.com/elastic/kibana/issues/69152)
+              await addSwimlaneToDashboardCallback(item);
+              setAddedDashboards({ ...addedDashboards, [item.id]: 'redirecting' });
+              await navigateToUrl(await dashboardService.getDashboardEditUrl(item.id));
+            },
+            type: 'icon',
+            icon: 'pencil',
+            enabled: (item) => !addedDashboards[item.id],
+          },
+        ],
+      },
+    ],
+    [addedDashboards, selectedSwimlanes]
+  );
+
+  const swimlaneTypeOptions = [
+    {
+      id: SWIMLANE_TYPE.OVERALL,
+      label: i18n.translate('xpack.ml.explorer.overallLabel', {
+        defaultMessage: 'Overall',
+      }),
+    },
+    {
+      id: SWIMLANE_TYPE.VIEW_BY,
+      label: i18n.translate('xpack.ml.explorer.viewByFieldLabel', {
+        defaultMessage: 'View by {viewByField}, up to {limit} rows',
+        values: { viewByField: viewBy, limit },
+      }),
+    },
+  ];
+
+  const redirectToDashboardInProgress = Object.entries(addedDashboards).find(
+    ([id, status]) => status === 'redirecting'
+  );
+
+  return (
+    <EuiOverlayMask>
+      <EuiModal onClose={onClose}>
+        <EuiModalHeader>
+          <EuiModalHeaderTitle>
+            <FormattedMessage
+              id="xpack.ml.explorer.dashboardsTitle"
+              defaultMessage="Attach swimlanes to dashboards"
+            />
+          </EuiModalHeaderTitle>
+        </EuiModalHeader>
+
+        <EuiModalBody>
+          {redirectToDashboardInProgress ? (
+            <EuiCallOut
+              size="s"
+              title={
+                <FormattedMessage
+                  id="xpack.ml.explorer.dashboardRedirectMessage"
+                  defaultMessage="Redirecting to the dashboard {dashboardTitle}"
+                  values={{
+                    dashboardTitle: dashboardItems.find(
+                      (item) => item.id === redirectToDashboardInProgress[0]
+                    )!.title,
+                  }}
+                />
+              }
+            />
+          ) : (
+            <>
+              <EuiButtonGroup
+                type="multi"
+                id="selectSwimlaneType"
+                name="selectSwimlaneType"
+                idToSelectedMap={selectedSwimlanes}
+                color="primary"
+                isFullWidth
+                legend={i18n.translate('xpack.ml.explorer.addToDashboard.selectSwimlanesLabel', {
+                  defaultMessage: 'Swimlanes to attach',
```

Comment:
Nit: Follow up to previous comment, maybe `Swimlanes to add`?

Reply:
In the recent changes, it is "Select swimlane view:"